src/Utils/Traceroute.ts

```diff
@@ -0,0 +1,236 @@
+import * as raw from 'raw-socket';
+import * as dgram from 'dgram';
+import { promises as dns } from 'dns';
+import { EventEmitter } from 'events';
+import { isIP } from 'net';
+
+export { Protocol } from 'raw-socket';
+
+type Options = {
+  amountProbes: number;
+  maximumHops: number;
+  timeoutInMillis: number;
+  reverseLookup: boolean;
+  protocol: number;
+  packetSize: number;
+  outStream: typeof process.stdout;
+};
+
+const defaultOptions: Options = {
+  amountProbes: 1,
+  maximumHops: 64,
+  timeoutInMillis: 3000,
+  reverseLookup: true,
+  protocol: raw.Protocol.UDP,
+  packetSize: 52,
+  outStream: process.stdout
+};
+
+export class Traceroute extends EventEmitter {
```

Comment:
Use composition instead of inheritance to prevent exposing details of implementation.